hadoop-ozone/cli-admin/src/main/java/org/apache/hadoop/hdds/scm/cli/container/ReconcileSubcommand.java

```diff
@@ -34,15 +45,183 @@
     versionProvider = HddsVersionProvider.class)
 public class ReconcileSubcommand extends ScmSubcommand {
 
-  @CommandLine.Parameters(description = "ID of the container to reconcile")
-  private long containerId;
+  @CommandLine.Mixin
+  private ContainerIDParameters containerList;
+
+  @CommandLine.Option(names = { "--status" },
+      defaultValue = "false",
+      fallbackValue = "true",
+      description = "Display the reconciliation status of this container's replicas")
+  private boolean status;
 
   @Override
   public void execute(ScmClient scmClient) throws IOException {
-    scmClient.reconcileContainer(containerId);
-    System.out.println("Reconciliation has been triggered for container " + containerId);
-    // TODO HDDS-12078 allow status to be checked from the reconcile subcommand directly.
-    System.out.println("Use \"ozone admin container info --json " + containerId + "\" to see the checksums of each " +
-        "container replica");
+    if (status) {
+      executeStatus(scmClient);
+    } else {
+      executeReconcile(scmClient);
+    }
+  }
+
+  private void executeStatus(ScmClient scmClient) throws IOException {
+    // Do validation outside the json array writer, otherwise failed validation will print an empty json array.
+    List<Long> containerIDs = containerList.getValidatedIDs();
+    try (SequenceWriter arrayWriter = JsonUtils.getSequenceWriter(System.out)) {
+      // Since status is retrieved using container info, do client side validation that it is only used for Ratis
+      // containers. If EC containers are given, print a  message to stderr and eventually exit non-zero, but continue
+      // processing the remaining containers.
+      int failureCount = 0;
+      for (Long containerID : containerIDs) {
+        if (!printReconciliationStatus(scmClient, containerID, arrayWriter)) {
+          failureCount++;
+        }
+      }
+      if (failureCount > 0) {
+        throw new RuntimeException("Failed to process reconciliation status for " + failureCount + " containers");
+      }
+    }
+    // Array writer will not add a newline to the end.
+    System.out.println();
+  }
+
+  private boolean printReconciliationStatus(ScmClient scmClient, long containerID, SequenceWriter arrayWriter) {
+    try {
+      ContainerInfo containerInfo = scmClient.getContainer(containerID);
+      if (containerInfo.getReplicationType() != HddsProtos.ReplicationType.RATIS) {
```

Comment:
We can add one more check here for `OPEN` containers so that we don't print 0 checksums for those. Also we don't reconcile `OPEN` containers, so no need to add that to the status.

Reply:
Do we want to fail the status check for open containers? I don't see much harm in just returning the 0 checksum. EC we definitely want to fail because the current `replicasMatch` check won't handle them correctly.

Reply:
There is no point in displaying 0 checksum for OPEN containers in Reconciliation Status as we don't reconcile OPEN containers. We don't have to throw an error, rather skip those container.

Reply:
I've added an error message for checking status of open containers and tests for this as well.